packages/shared/src/models/examples.ts

```diff
@@ -0,0 +1,59 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface Category {
+  id: string;
+  name: string;
+  description?: string;
+}
+
+// validator for certain types arm / amd64
+export type ArchitectureType = 'arm64' | 'amd64';
+
+export interface Example {
+  id: string;
+  name: string;
+  categories: string[];
+  description: string;
+  repository: string;
+
+  // List of available architectures to pull wtih the image.
+  architectures?: ArchitectureType[];
+
+  // container image to pull, some may not be available... so make this optional.
+  image?: string;
+
+  // tag, else we will assume 'latest'
+  tag?: string;
+
+  // Optional reference version for the example
```

Comment:
Nice this one says Optional, but it's the only one that doesn't say what the default behaviour is. All of these optional parameters are phrased a bit differently, would be nice to make them all consistent. e.g. like 'X. Defaults to Y', don't use 'will assume'.

Reply:
Honestly it would be good just to remove this as "tag" would be used for the image, and we are not pulling / using references yet for git... Could revisit in the future if we add the functionality to download the Containerfile locally / git checkout.